File: typescript/src/menu.ts
import gradient from 'gradient-string';
import inquirer from 'inquirer';
import { CONFIG, isConfiguredUrl } from './config.js';

export interface MenuChoice {
	name: string;
	value: string;
}

export const getMenuOptions = () => [
	{
		type: 'list',
		name: 'action',
		message: gradient.cristal('What would you like to do?'),
		choices: [
			isConfiguredUrl(CONFIG.urls.email)
				? { name: `📧  ${gradient.passion('Send an Email')}`, value: 'email' }
				: null,
			isConfiguredUrl(CONFIG.urls.resume)
				? { name: `📥  ${gradient.morning('View Resume')}`, value: 'viewResume' }
				: null,
			isConfiguredUrl(CONFIG.urls.portfolio)
				? { name: `🌐  ${gradient.teen('Visit Portfolio')}`, value: 'viewPortfolio' }
				: null,
			isConfiguredUrl(CONFIG.urls.github)
				? { name: `💻  ${gradient.atlas('View GitHub')}`, value: 'viewGitHub' }
				: null,
			isConfiguredUrl(CONFIG.urls.linkedin)
				? { name: `💼  ${gradient.fruit('View LinkedIn')}`, value: 'viewLinkedIn' }
				: null,
			isConfiguredUrl(CONFIG.urls.twitter)
				? { name: `🐦  ${gradient.cristal('View Twitter')}`, value: 'viewTwitter' }
				: null,
			{ name: gradient.rainbow('🚪  Exit'), value: 'quit' },
		].filter((choice): choice is MenuChoice => Boolean(choice)),
	},
];

export const promptUser = async (): Promise<{ action: string }> => {
	return inquirer.prompt(getMenuOptions());
};

File: python/src/carlosferreyra/menu.py
"""Interactive menu system."""

import inquirer
from rich.console import Console

from .config import CONFIG, is_configured_url

console = Console()


def get_menu_choices():
    """Get the menu choices for the interactive prompt."""
    choices = []

    if is_configured_url(CONFIG.urls.email):
        choices.append(("📧  Send an Email", "email"))

    if is_configured_url(CONFIG.urls.resume):
        choices.append(("📥  View Resume", "view_resume"))

    if is_configured_url(CONFIG.urls.portfolio):
        choices.append(("🌐  Visit Portfolio", "view_portfolio"))

    if is_configured_url(CONFIG.urls.github):
        choices.append(("💻  View GitHub", "view_github"))

    if is_configured_url(CONFIG.urls.linkedin):
        choices.append(("💼  View LinkedIn", "view_linkedin"))

    if is_configured_url(CONFIG.urls.twitter):
        choices.append(("🐦  View Twitter", "view_twitter"))

    choices.append(("🚪  Exit", "quit"))
    return choices


def prompt_user() -> str:
    """Prompt user for menu selection."""
    choices = get_menu_choices()

    questions = [
        inquirer.List(
            "action",
            message="What would you like to do?",
            choices=choices,
            carousel=True,
        )
    ]

    try:
        answers = inquirer.prompt(questions)
        return answers["action"] if answers else "quit"
    except (KeyboardInterrupt, EOFError):
        console.print("\n[yellow]👋 Thanks for stopping by! Have a great day![/yellow]")
        return "quit"
